shipyard-controller/handler/sequencedispatcher.go

```diff
@@ -57,24 +62,34 @@ func NewSequenceDispatcher(
 }
 
 func (sd *SequenceDispatcher) Add(queueItem models.QueueItem) error {
-	// try to dispatch the sequence immediately
-	if err := sd.dispatchSequence(queueItem); err != nil {
-		if err == ErrSequenceBlocked {
-			// if the sequence is currently blocked, insert it into the queue
-			if err2 := sd.sequenceQueue.QueueSequence(queueItem); err2 != nil {
-				return err2
-			}
-		} else if err == ErrSequenceBlockedWaiting {
-			// if the sequence is currently blocked and should wait, insert it into the queue
-			if err2 := sd.sequenceQueue.QueueSequence(queueItem); err2 != nil {
-				return err2
+
+	if os.Getenv(EnvVarDisableLeaderElection) == "true" {
```

Comment:
Please do not read environment variables out of the blue right in the middle of the code....It is hard to test. Also, the `SequenceDispatcher` does not need to know anything about the concept of "leader election" it can just operate in two different operational "modes" as it seems. I would propose to introduce a kind of distinction using an enum:

```go
type SDMode int

const (
   // document what this mode is supposed to do
   SDModeRW SDMode = iota
   // document what that mode is supposed to do
   SDModeW 
)
```

And store the SD mode as a field of the `SequenceDispatcher`